### PR TITLE
[Merged by Bors] - feat: port FieldTheory.IsAlgClosed.AlgebraicClosure

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by xroblot** [comment]
I pushed a couple of fixes but I have to sleep now

**Comment by riccardobrasca** [comment]
I am having a look at this.

**Comment by nick-kuhn** [comment]
This is now basically done except for `Step.isIntegral`, where one would have to decide whether to increase the Heartbeats to 2 mil. or change something.

**Comment by nick-kuhn** [comment]
It actually did time out for me with 2 mil...? I'll make an attempt to change it.

File: Mathlib/FieldTheory/IsAlgClosed/AlgebraicClosure.lean [hunk 1]
```diff
@@ -130,42 +133,50 @@ instance AdjoinMonic.algebra : Algebra k (AdjoinMonic k) :=
   (toAdjoinMonic k).toAlgebra
 #align algebraic_closure.adjoin_monic.algebra AlgebraicClosure.AdjoinMonic.algebra
 
-theorem AdjoinMonic.algebraMap : algebraMap k (AdjoinMonic k) = (Ideal.Quotient.mk _).comp C :=
-  rfl
+theorem AdjoinMonic.algebraMap : algebraMap k (AdjoinMonic k) = (Ideal.Quotient.mk _).comp
+  (C : k →+* MvPolynomial (MonicIrreducible k) k) := rfl
```
Comment (jcommelin) [inline Mathlib/FieldTheory/IsAlgClosed/AlgebraicClosure.lean:137]:
```suggestion
-- Porting note: In the statement, the type of `C` had to be made explicit.
theorem AdjoinMonic.algebraMap : algebraMap k (AdjoinMonic k) = (Ideal.Quotient.mk _).comp
  (C : k →+* MvPolynomial (MonicIrreducible k) k) := rfl
```

File: Mathlib/FieldTheory/IsAlgClosed/AlgebraicClosure.lean [hunk 2]
```diff
@@ -255,31 +327,45 @@ instance algebraOfStep (n) : Algebra (Step k n) (AlgebraicClosure k) :=
...
+  dsimp [toStepOfLE]
+  rw [← hx]
+  change Ring.DirectLimit.of (Step k) (toStepOfLE' k) (n + 1) (_) =
+      Ring.DirectLimit.of (Step k) (toStepOfLE' k) n x
+  convert Ring.DirectLimit.of_f n.le_succ x
+--Porting Note: Original proof timed out at 2 mil. Heartbeats. The problem was likely
+--in comparing `toStepOfLE'` with `toStepSucc`. In the above, I made some things more explicit
+  -- RingHom.ext fun x =>
+  --   show Ring.DirectLimit.of (Step k) (fun i j h => toStepOfLE k i j h) _ _ = _ by
+  --     convert Ring.DirectLimit.of_f n.le_succ x; ext x; exact (Nat.leRecOn_succ' x).symm
```
Comment (jcommelin) [inline Mathlib/FieldTheory/IsAlgClosed/AlgebraicClosure.lean:344]:
```suggestion
  -- Porting Note: Original proof timed out at 2 mil. Heartbeats. The problem was likely
  -- in comparing `toStepOfLE'` with `toStepSucc`. In the above, I made some things more explicit
  -- Original proof:
  -- RingHom.ext fun x =>
  --   show Ring.DirectLimit.of (Step k) (fun i j h => toStepOfLE k i j h) _ _ = _ by
  --     convert Ring.DirectLimit.of_f n.le_succ x; ext x; exact (Nat.leRecOn_succ' x).symm
```

**Comment by ChrisHughes24** [comment]
bors merge

**Comment by riccardobrasca** [comment]
It's sad that it is so slow, but we can always improve it later, thanks!

bors merge

**Pull request closed without merging**